CRM/Contact/BAO/Query.php

```diff
@@ -5779,13 +5779,24 @@ public function getOperator() {
    * @param $having
    */
   public function filterRelatedContacts(&$from, &$where, &$having) {
-    static $_rTypeProcessed = NULL;
-    static $_rTypeFrom = NULL;
-    static $_rTypeWhere = NULL;
-
-    if (!$_rTypeProcessed) {
-      $_rTypeProcessed = TRUE;
-
+    static $_rTempCache = array();
```

Comment:
I have not placed these code for the unit-test. This is the fix. Maybe changing it to \Civi::statics has advantages i don't know.

Reply:
@hosseinamin yes - I realised that the use of $arg_sig is the test & I think it will fix the bug as described once we get the tests sorted. And thank you for working on this!

You may find that once your test running in the phpunit Continuous integration the type of static might need changing. The reason is that the Civi::statics cache is flushed between tests so you might see unexpected fails elsewhere in the suite.

We used to have 2 suites of tests - the phpunit ones  & the selenium / webtests ones. The latter still exist but they are too full of false negatives to run whereas the former run against every PR that is submitted and are run nightly on different php / mysql combos